### ISSUE TITLE
`Style/RedundantConstantBase` when working in `*.jbuilder` files

### ISSUE THREAD

**Issue by stdedos** [rubocop/rubocop#13377]
## Expected behavior

RuboCop should not suggest removing the `::` when resolving a constant explicitly from the top-level namespace, when "it is not sure"  what the top-level namespace is 😕 

```ruby
json.cache! ['renderers/index', User.current.id, ::Renderer.soft_cache_key] do
  # Expected to use the top-level Renderer
end
```

## Actual behavior

RuboCop flags the `::` as redundant and suggests removing it, which could cause the constant `Renderer` to be resolved incorrectly if there's a similarly named constant in the current scope.

```console
$ rubocop --debug a/views/r/i.json.jbuilder
a/views/r/i.json.jbuilder:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
json.cache! ['r/i', User.current.id, Renderer.soft_cache_key] do
^
a/views/r/i.json.jbuilder:1:50: C: [Corrected] Style/RedundantConstantBase: Remove redundant ::.
json.cache! ['r/i', User.current.id, ::Renderer.soft_cache_key] do
                                     ^^
 1/1 file |======================= 100 =======================>| Time: 00:00:01 

1 file inspected, 2 offenses detected, 1 offense corrected, 1 more offense can be corrected with `rubocop -A`
```

However, it seems that 

```ruby
JSON::ParserError:
  unexpected token at 'ActionView::Template::Error: undefined method `soft_cache_key' for ActionView::Renderer:Class
    /xxxxxx/xxxxxx/xxxxxxx-xxxxx/xxxxxxx/xxx/views/xxxxxxxx/index.json.jbuilder:1:in `_xxx_xxxx_xxxxxxxxx_xxxxx_xxxx_jbuilder___1521783346_10232'
    org/jruby/RubyKernel.java:2247:in `public_send'
    /xxxxxx/xxxxxx/xxxxxxx-xxxxx/xxxx/xxxx/.xxxxx-xxxxx/9.4.8.0/gems/actionview-7.1.4.1/lib/action_view/base.rb:263:in `_run'
    ...
```

`ActionView` _apparently_ has a `Renderer`, which takes precedence over "the top-level namespace".

## Steps to reproduce the problem

This is extremely important! Providing us with a reliable way to reproduce a problem will expedite its solution.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ [bundle exec] rubocop -V
1.67.0 (using Parser 3.3.5.0, rubocop-ast 1.32.3, analyzing as Ruby 3.3, running on ruby 3.3.5) [x86_64-linux]
  - rubocop-performance 1.22.1
  - rubocop-rspec 3.1.0
```

**Comment by dvandersluis** [comment]
RuboCop cannot know that your file is being executed in another context, as Rails is doing here. It seems to me that the best solution is to add `**/*.jbuilder`  to `Exclude` for `Style/RedundantConstantBase`.

**Comment by stdedos** [comment]
... maybe at the Rubocop side?

**Comment by bbatsov** [comment]
I prefer to keep the default configuration as universal as possible and generally avoid tool-specific items there. That's why similar to the reason why `rubocop-rails` was extracted out of RuboCop in the past.

**Comment by stdedos** [comment]
Of course!

... Should plain rubocop _avoid_ to touch jbuilder files? Or can `rubocop-rails` "change" the default config of a "standard" cop?

**Comment by bbatsov** [comment]
That's a fair question without a clear answer, as you can't be sure how exactly they get used. Technically speaking you don't have to be in Rails app to do something with them, that's why they were added in the default config in the first place.
